dashboard/lib/state/build.dart

```diff
@@ -108,30 +117,43 @@ class BuildState extends ChangeNotifier {
   /// Start a fixed interval loop that fetches build state updates based on [refreshRate].
   void _startFetchingStatusUpdates() {
     assert(refreshTimer == null);
+    _fetchBranches();
+    _fetchRepos();
     _fetchStatusUpdates();
     refreshTimer = Timer.periodic(refreshRate, _fetchStatusUpdates);
   }
 
+  /// Request the latest [branches] from [CocoonService].
+  Future<void> _fetchBranches() async {
+    final CocoonResponse<List<String>> response = await cocoonService.fetchFlutterBranches();
+
+    if (response.error != null) {
+      _errors.send('$errorMessageFetchingBranches: ${response.error}');
+    } else {
+      _branches = response.data;
+      notifyListeners();
+    }
+  }
+
+  /// Request the latest [repos] from [CocoonService].
+  Future<void> _fetchRepos() async {
+    final CocoonResponse<List<String>> response = await cocoonService.fetchRepos();
+    if (response.error != null) {
+      _errors.send('$errorMessageFetchingBranches: ${response.error}');
+    } else {
+      _repos = response.data;
+      notifyListeners();
+    }
+  }
+
   /// Request the latest [statuses] and [isTreeBuilding] from [CocoonService].
   ///
   /// If fetched [statuses] is not on the current branch it will be discarded.
   Future<void> _fetchStatusUpdates([Timer timer]) async {
     await Future.wait<void>(<Future<void>>[
-      () async {
-        final CocoonResponse<List<String>> response = await cocoonService.fetchFlutterBranches();
```

Comment:
This doesn't need to be run every 30 seconds as this is almost never changes (order of once a month)